tests/simulator/py/fluid_init.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+#
+# formatted with black
+
+from tests.simulator.py import InitValueValidation
+
+import phare.pharein as ph, numpy as np
+from phare.pp.diagnostics import Diagnostics
+
+
+class FluidInitValidation(InitValueValidation):
+
+    pop_diag_file = lambda fn_name: "ions_pop_ions_${POP}_" + fn_name
+
+    def test_1d(self):
+        from phare.pp.diagnostics import _Fluid
+        from tests.diagnostic import dump_all_diags
+
+        diag_out_dir = "phare_outputs/fluid_init"
+        dic = InitValueValidation.diag_options(diag_out_dir)
+        dic.update(
+            {
+                "diags_fn": lambda model: dump_all_diags(model.populations),
+                "populations": {
+                    "beta": {"nbr_part_per_cell": 10000, "init": {"seed": 133337}},
+                    "gamma": {"nbr_part_per_cell": 100000, "init": {"seed": 1333337}},
+                },
+            }
+        )
+
+        diags = self._simulate_diagnostics(dim=1, interp=1, input=dic)[_Fluid.__name__]
+        self._checkFluidFromDensityFN(diags)
+        self._checkFluidFromFluxFN(diags)
+        self._checkIons(diags)
+
+    def _sum_pop_patches(self, pop_patches, dataset):
+        return np.sum(
+            [
+                self._aggregate_physical_patch_value(patches, True)[dataset]
+                for pop, patches in pop_patches.items()
+            ],
+            axis=0,
+        )
+
+    def _checkIons(self, fluids):
+        tolerance = 1e-6
+        sim, model_dict = ph.globals.sim, ph.globals.sim.model.model_dict
+
+        pop_densities, pop_fluxs = [
+            self._get_contig_popdata_4_level0_ions_type(
+                fluids, FluidInitValidation.pop_diag_file(fn_name=name)
+            )
+            for name in ["density", "flux"]
+        ]
+
+        sum_pop_density = self._sum_pop_patches(pop_densities, "density")
+        ions_density = self._get_contig_data_4_level0_ions_type(fluids, "ions_density")[
+            "density"
+        ]
+        np.testing.assert_allclose(
+            sum_pop_density, ions_density, rtol=tolerance,
+        )
+
+        sum_pop_flux = {
+            key: self._sum_pop_patches(pop_fluxs, key) for key in ["x", "y", "z"]
+        }
+        ions_bulkVelocity = self._get_contig_data_4_level0_ions_type(
+            fluids, "ions_bulkVelocity"
+        )
+        for key, val in ions_bulkVelocity.items():
+            np.testing.assert_allclose(
+                sum_pop_flux[key] / sum_pop_density, val, rtol=tolerance
+            )
+
+    def _checkFluidFromFluxFN(self, fluids, fn_name="flux"):
+        sim, model_dict = ph.globals.sim, ph.globals.sim.model.model_dict
+        pops = self._get_contig_popdata_4_level0_ions_type(
```

Comment:
what is `pops` ? rename plz is that  `population_fluxes`?

Reply:
pop_fluxes?